sda-commons-client-jersey/README.md

```diff
@@ -168,6 +168,49 @@ executorService.submit(transferRequestContext(() -> {
 })).get();
 ```
 
+## HTTP Client Configuration and Proxy Support
+
+Each client can be configured with the standard
+[Dropwizard configuration](https://www.dropwizard.io/en/latest/manual/configuration.html#man-configuration-clients-http).
+Please note that this requires that there is a property in your Application's `Configuration` class. 
+
+```java
+import org.sdase.commons.client.jersey.HttpClientConfiguration;
+
+public class MyConfiguration extends Configuration {
+   private HttpClientConfiguration myClient;
+
+   public HttpClientConfiguration getMyClient() {
+     return myClient;
+   }
+
+   public void setMyClient(HttpClientConfiguration myClient) {
+     this.myClient = myClient;
+   }
+}
+```
+
+```java
+Client client = clientFactory.platformClient(configuration.getMyClient()).buildGenericClient("test");
+```
+
+```yaml
+myClient:
+  timeout: 500ms
+  proxy:
+    host: 192.168.52.11
+    port: 8080
+    scheme : http
+```
+
+> Tip: There is no need to make all configuration properties available as environment variables.
+> Seldomly used properties can always be configured using [System Properties](https://www.dropwizard.io/en/latest/manual/core.html#man-core-configuration).
+
+This configuration can be used to configure a proxy server if needed.
+Use this if all clients should use individual proxy configurations.
+
+In addition, each client consumes the standard [proxy system properties](https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html#Proxies).
+This can be helpful when all clients in an Application should use the same proxy configuration (this includes the clients that are used by the [`sda-commons-server-auth` bundle](../sda-commons-server-auth).
```

Comment:
Do the specific settings of the `JerseyClientConfiguration` override the global _proxy system settings_? Can we document more clearly, if individual settings and global settings are exclusive or can be used at the same time? Can global settings be used when the proxy setting of a client is defined in the `config.yaml` or is it required to use the individual settings in this case? Do things get complicated with this many options?

Reply:
Global settings override specific settings. (In fact, it seems that the module for global settings _always_ overrides the specific settings). I'll think about how to handle this.

Reply:
We could change this to: Specific settings override global settings. We can use the same check logic that Dropwizard uses to check whether the `NonProxyListProxyRoutePlanner` should be used to not add the `SystemDefaultRoutePlanner` in that case.

However, I get problems when one wants to set the non-proxy-hosts via the system property `dw.client.proxy.nonProxyHosts`. Since it is an array, it only works in case the property exists as an array before in the config file, which isn't the wanted behavior. We won't provide the config settings as part of the default `config.yml` of all our services. But I could update the documentation to highlight that it is preferred to use the global proxy settings.

Reply:
I changed the logic: If there is no specific proxy configuration for a client, it defaults to the default java properties. Service creators should not add any proxy related configuration to their `config.yml`. If needed, this should be done via system properties (would be nice if dropwizard would also support environment variables).